### PR TITLE
SPARK-20433 Bump jackson from 2.6.5 to 2.6.7.1

### PR DESCRIPTION
**Pull request by ash211**
This brings in a security fix for CVE-2017-7525 in the jackson-databind library, which Spark uses.

When releasing this patch, upstream released a 2.6.7.1 for jackson-databind but not a corresponding 2.6.7.1 for the rest of jackson, so those only go up to 2.6.7

This requires splitting the version variable in /pom.xml

### CODE REVIEW

**Comment by srowen** [comment]
@ash211 I still don't think this should be described as a CVE fix, because it doesn't appear to affect Spark. It is, however, going to be necessary for Scala 2.12. To me that's the most real motivation here.

I tried this change in my scala 2.12 branch and found I needed to do a bit more:
https://github.com/apache/spark/pull/18645/files#diff-07fec5b101338c859a2f08f3801a5bd8

`jackson-module-scala` also needed to update to 2.6.7.1 because that's the first version that has a 2.12 artifact. So I used the 'databind' property there as well. Ideally, just make that change here too and then it aligns with what 2.12 needs anyway.

File: sql/core/pom.xml [hunk 1]
```diff
@@ -101,7 +101,7 @@
     <dependency>
       <groupId>com.fasterxml.jackson.core</groupId>
       <artifactId>jackson-databind</artifactId>
-      <version>${fasterxml.jackson.version}</version>
+      <version>${fasterxml.jackson.databind.version}</version>
```
Comment (zsxwing) [inline sql/core/pom.xml]: hm, why do we need a version number here? Could you just remove it?

Comment (srowen) [inline sql/core/pom.xml]: Oh yeah forgot I made that change too. Agree

**Comment by srowen** [comment]
Ping @ash211 or should I take this over?

**Comment by ash211** [comment]
@srowen sorry for not picking up on this -- thanks for pushing it over the finish line in your PR!

**Pull request closed without merging**